alacritty/src/event.rs

```diff
@@ -1327,21 +1339,167 @@ impl Processor {
         }
     }
 
+    /// Create the GL display.
+    fn create_gl_display(
+        raw_display_handle: RawDisplayHandle,
+        _raw_window_handle: Option<RawWindowHandle>,
+    ) -> GlutinResult<GlutinDisplay> {
+        #[cfg(target_os = "macos")]
+        let preference = DisplayApiPreference::Cgl;
+
+        #[cfg(windows)]
+        let preference = DisplayApiPreference::Wgl(Some(_raw_window_handle.unwrap()));
+
+        #[cfg(all(feature = "x11", not(any(target_os = "macos", windows))))]
+        let preference = DisplayApiPreference::GlxThenEgl(Box::new(unix::register_xlib_error_hook));
+
+        #[cfg(all(not(feature = "x11"), not(any(target_os = "macos", windows))))]
+        let preference = DisplayApiPreference::Egl;
```

Comment:
Why do you have to do all this manually? The goal for `glutin` is to provide an abstraction, right? Couldn't it provide a `PlatformApiPreference` that automatically works?

Reply:
Because some requires things like `unix::register_xlib_error_hook` which may not be present at all, or `RawWindowHandle` without a reason. I thought how to abstract it in a way that you can still use glutin for example without WGL and use that interface, but I have no real ideas...

If you have any let me know.

If we could have preferences without extra data it could provide something like that.